Pin nuget vcr cassette names

mctofu opened the pull request:
When we have VCR enabled tests without a provided description like
```
its([:version]) { is_expected.to eq(version_class.new("7.3.0")) }
```

The VCR library will name the cassette [based on the `:scoped_id`](https://github.com/vcr/vcr/blob/5c0ded824dd8b1f9d50ab5d48a53be065b94ece8/lib/vcr/test_frameworks/rspec.rb#L10-L15) of the test. That id will change if the test is moved around or other tests are added before it. Our VCR config allows recording a new cassette which means a test with a changed id is silently fetching the latest data and will eventually fail when the latest data no longer matches our test expectations.

This PR fixes occurrences of this problem in the nuget tests (which are currently causing a failure) by adding a description to the test. We can address other ecosystems & the VCR config in a later PR.

mctofu commented:
Some other potentially flaky cassettes:
```
% find . -name "1_*.yml"
./terraform/spec/fixtures/vcr_cassettes/Dependabot_Terraform_MetadataFinder/_source_url/with_a_provider/1_2_3_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_PullRequestCreator_MessageBuilder/_commit_message/for_a_repo_that_uses_gitmoji_commits/1_3_4_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_PullRequestCreator_MessageBuilder/_commit_message/for_a_repo_that_uses_gitmoji_commits/with_a_security_vulnerability_fixed/1_3_4_2_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_PullRequestCreator_MessageBuilder/_commit_message/1_3_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_PullRequestCreator_MessageBuilder/_pr_message/for_an_application/with_multiple_git_source_requirements/1_2_1_11_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_PullRequestCreator_MessageBuilder/_pr_name/for_an_application/with_multiple_git_source_requirements/1_1_1_5_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_MetadataFinders_Base_ReleaseFinder/_releases_text/with_a_github_repo/with_releases/when_the_release_is_present/updating_from_uknown_previous_release_to_new_release/1_2_1_1_1_4_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_MetadataFinders_Base_ReleaseFinder/_releases_text/with_a_github_repo/with_releases/when_the_release_is_present/updating_from_no_previous_release_to_new_release/1_2_1_1_1_4_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_MetadataFinders_Base_ChangelogFinder/_changelog_url/with_a_github_repo/for_a_git_dependency_with_multiple_sources/1_1_1_7_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_MetadataFinders_Base_ChangelogFinder/_changelog_text/with_a_github_repo/with_a_changelog/for_a_git_dependency_with_multiple_sources/when_the_ref_has_changed/1_2_1_1_6_1_1.yml
./common/spec/fixtures/vcr_cassettes/Dependabot_MetadataFinders_Base_ChangelogFinder/_changelog_text/with_a_github_repo/with_a_changelog/for_a_git_dependency_with_multiple_sources/when_there_s_a_new_ref/1_2_1_1_6_1_1.yml
./github_actions/spec/fixtures/vcr_cassettes/Dependabot_GithubActions_UpdateChecker/_can_update_/given_a_dependency_with_a_tag_reference/with_a_dependency_that_has_a_latest_requirement_and_a_valid_version/1_2_2_6_1.yml
./github_actions/spec/fixtures/vcr_cassettes/Dependabot_GithubActions_UpdateChecker/_updated_requirements/with_multiple_requirement_sources/1_5_4_1.yml
./github_actions/spec/fixtures/vcr_cassettes/Dependabot_GithubActions_UpdateChecker/_latest_version/given_a_dependency_with_multiple_git_refs/1_3_4_1.yml
```

jurre commented:
I'd expect a new cassette to have been added? Are we still recording a new one every time CI runs now?

mctofu commented:
> I'd expect a new cassette to have been added? Are we still recording a new one every time CI runs now?

Nope. The currently failing test was last recorded with `1_1_15_1_1.yml`. Something else changed in this file and that test now expects `1_1_16_1_1.yml` which worked a bit but now fails because there's been a new version of the tested dependency released. I renamed `1_1_15_1_1.yml` to `returns_the_expected_version.yml` so the test is back to using the original recording.

I should have mentioned in the description I also found some other unused cassettes and deleted them.
